components/org.wso2.carbon.securevault/src/test/java/org/wso2/carbon/securevault/DefaultMasterKeyReaderTest.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.securevault;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import org.wso2.carbon.securevault.exception.SecureVaultException;
+import org.wso2.carbon.securevault.model.masterkey.MasterKeyConfiguration;
+import org.wso2.carbon.securevault.reader.DefaultMasterKeyReader;
+import org.wso2.carbon.securevault.utils.ClassUtils;
+import org.wso2.carbon.securevault.utils.EnvironmentUtils;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.introspector.BeanAccess;
+import org.yaml.snakeyaml.nodes.Tag;
+import org.yaml.snakeyaml.representer.Representer;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Unit tests class for org.wso2.carbon.securevault.reader.DefaultMasterKeyReader.
+ * Priority 1: System property
+ * Priority 2: Environment variable
+ * Priority 3: master-keys.yml
+ *
+ * @since 5.2.0
```

Comment:
As mentioned above this should change to the carbon-secvault version, this file is being introduced. Similarly, we need to change all the class "@Since" version, as this is moved from kernel to carbon-secvault. Please check and change accordingly.